File: test69_out/pollen.event/Events/Events.c

/*
 * ======== string literals ========
 */

static const char pollen_event_Events__s__fileName[] = "Events.p";
static const char pollen_event_Events__s__2[] = "in Events(), numEvents=";
static const char pollen_event_Events__s__5[] = "in registerEvent(), Event Registered: ";
static const char pollen_event_Events__s__8[] = "Events";
static const char pollen_event_Events__s__4[] = "in registerEvent(), numEvents=";
static const char pollen_event_Events__s__1[] = "Event";
static const char pollen_event_Events__s__6[] = " \n";
static const char pollen_event_Events__s__7[] = "numEvents=";
static const char pollen_event_Events__s__0[] = "in Event new_host\n";
static const char pollen_event_Events__s__3[] = "\n";

/*
 * ======== function definitions ========
 */

void pollen_event_Events_Event_setHandler__E( pollen_event_Events_Event_ this, pollen_event_Events_HP_handler h ) {
    (this->handler) = (h);
}

void pollen_event_Events_test__E( pollen_event_Events_Event_ e ) {
    virtual_mcu_ConsolePrint_printStr__E(pollen_event_Events__s__7);
    virtual_mcu_ConsolePrint_printUint__E((pollen_event_Events_numEvents__V));
    e->handler();
}

pollen_event_Events_Event_ pollen_event_Events_Event_new___E( pollen_event_Events_Event_ this, pollen_event_Events_HP_handler h ) {
    (this->handler) = (h);
    return this;
}

void pollen_event_Events_targetInit__I() {
}

File: test69_out/pollen.event/Events/Events.js

/*
 * ======== MODULE pollen.event.Events ========
 */

var $$c = function() {
    const Events = {};
    Events.Events = Events
    Events.$name = 'pollen.event.Events';
    Events.pollen$used = 0;
    Events.CP = $units['virtual.mcu.ConsolePrint'];
    Events.Event$$id = 1;
    Events.Event = function(cn) {
        this.$$id = Events.Event$$id++;
        this.$$cname = cn ? cn : 'pollen_event_Events_Event__'+this.$$id+'__S';
        this.$$tname = 'pollen_event_Events_Event';
        this.$$uname = 'pollen.event.Events';
        this.handler = 0;
    }
    Events.Event.prototype = new $$Struct('pollen.event.Events.Event');
    Events.Event.prototype.$$isAggFld = {};
    Events.Event.prototype.$$isAggFld.handler = false;
    Events.Event.prototype.new_host = function( h ) {
        var $$text = '';
        this.$$privateInit();
        if (arguments.length == 1) {
                $$printf("in Event new_host\n");
                Events.registerEvent();
                (this.handler) = (h);
        }
        this.$$hostInitLastPass();
        return this;
    }
    Events.Event.prototype.setHandlerOnHost = function( h ) {
        var $$text = '';
        (this.handler) = (h);
        this.$$hostInitLastPass();
        return $$text;
    }
    Events.Event.prototype.$$privateInit = function() {
    }
    Events.Event.prototype.$$hostInitLastPass = function() {
    }
    Events.targetInit = new $$Ref('pollen_event_Events_targetInit__I');
    Events.test = new $$Ref('pollen_event_Events_test__E');
    Events.$$hostInit = function() {
        var $$text = '';
        $$printf("in Events(), numEvents=", (Events.numEvents), "\n");
        return $$text;
    }
    Events.registerEvent = function() {
        var $$text = '';
        $$printf("in registerEvent(), numEvents=", (Events.numEvents), "\n");
        ++(Events.numEvents);
        $$printf("in registerEvent(), Event Registered: ", (Events.numEvents), " \n");
        return $$text;
    }
    Events.pollen__uses__ = function() {
        $$bind($units['virtual.mcu.ConsolePrint'], 'pollen$used', true);
    }
    Events.$$privateInit = function() {
        Events.numEvents = 0;
    }
    return Events;
}
var $$u = $$c();
$units['pollen.event.Events'] = $$u;
$units.push($$u);

